tests/v2/waku_filter_v2/test_waku_filter_protocol.nim

```diff
@@ -237,3 +235,64 @@ suite "Waku Filter - handling subscribe requests":
       response3.statusCode == 200
       response3.statusDesc.get() == "OK"
 
+suite "Waku Filter - subscription maintenance":
+
+  asyncTest "simple maintenance":
+    # Given
+    let
+      switch = newStandardSwitch()
+      wakuFilter = newTestWakuFilter(switch)
+      peerId1 = PeerId.random().get()
+      peerId2 = PeerId.random().get()
+      peerId3 = PeerId.random().get()
+      filterSubscribeRequest = createRequest(
+        filterSubscribeType = FilterSubscribeType.SUBSCRIBE,
+        pubsubTopic = some(DefaultPubsubTopic),
+        contentTopics = @[DefaultContentTopic]
+      )
+
+    # When
+    switch.peerStore[ProtoBook][peerId1] = @[WakuFilterPushCodec]
+    switch.peerStore[ProtoBook][peerId2] = @[WakuFilterPushCodec]
+    switch.peerStore[ProtoBook][peerId3] = @[WakuFilterPushCodec]
+    discard wakuFilter.handleSubscribeRequest(peerId1, filterSubscribeRequest)
```

Comment:
maybe nitpick?

`require wakuFilter.handleSubscribeRequest(peerId1, filterSubscribeRequest).isOk`

Reply:
Agree. Done in [9105f88](https://github.com/waku-org/nwaku/pull/1600/commits/9105f88a45688599ea8ee580a873aa17a3af21ac)